main/2023/CG-10.md

```diff
@@ -113,4 +113,1850 @@ Use this space to add any agenda items that need to be scheduled, and scheduling
+
+AR: They might have to do pass more than just an RTT. IT’s not sufficient, but it is a building block
+
+CR: It’s a good start but I need to think more.
+
+RT: For something like ocaml to use this, because it needs to downcast at some type. Every polymorphic function will also be gicen the rtt for it, you will then have to construct the rtt on the fly, and then you have to do the type construction dynamically, is this worth it? <TODO: RT to fix>
+
+AR: The whole point is not to pass an RTT everywhere. Basically when you compile something like OCaml with a type repr you have type_eq anyway. And only the places where you consume a value you need a cast but at that point you usually know the type you are casting to. In addition this allows a polymorphic call has a more concrete type, you might also get out a more concrete type when it gets back to you.
+
+RT: I can’t construct a pair value of alpha alpha, without knowing the rtt for tag (?)
+
+AR: That is correct. The other feature we need is a way of constructing/allocating values without RTT. So if you never want to cast them, they don’t need to carry RTT. A good question hwo much of your code would fall into the section of things where you never need RTT but it’s worth investigating whether it gives some benefit in circumstances. 
+
+ZB: Both features look like we would want to use them in Kotlin, more request - how about having import/export of unsigned numbers, could be used to provide better APIs and to go abc and forth from the host. 
+
+AR: What do you mean importing unsigned numbers?
+
+ZB: Allowing using u8 in import/export section
+
+AR: Not just about imports/exports, we don’t have small integers. Once we have them you can import/export them as an immutable global but it is not something we have included.
+
+ZB: Another thing about typed parameters is about covariant and  invariant things and different bounds for typed parameters
+
+AR: That’s a riddle. Something we could do is variance and polarities as they are sometimes called and plausible but not entirely straightforward and another added complexity. There is a point of diminishing returns but we have to figure out where that is, so for polarities we have to figure out where that is but nothing technically preventing that.
+
+BT: Scala has a long complicated history of type parameters, their experiments of reified ?? <BT to fix>
+
+AR: maybe that’s why i'm bringing it up, worth investigating, I don’t have an answer. My gut is it is.
+
+<we have no idea>: Just wanted to add what Ben keeps saying, I’m skeptical of type parameters at all because of the type varieties there are across the languages. 
+
+AR: most languages allow you to specify variance much, scala does, lets not talk about Java, doesn’t matter anyway. Some modern OOO languages have it, Ocaml doesn’t count
```

Comment:
```suggestion
AR: most languages don't allow you to specify variance much, scala does, lets not talk about Java, doesn’t matter anyway because type erasure. Some modern OOO languages have it, Ocaml doesn’t count
```